### PR TITLE
Support highlight within queryDSL field #268

### PR DESCRIPTION
**Pull request by anatol-sialitski**

### CODE REVIEW

File: src/main/resources/lib/guillotine/query/content-api.js [hunk 1]
```diff
@@ -224,21 +224,27 @@ function createContentApiType(context) {
...
                     first: graphQlLib.GraphQLInt,
                     aggregations: graphQlLib.list(context.types.aggregationInputType),
+                    highlight: context.types.highlightInputType,
                     sort: graphQlLib.list(context.types.sortDslInputType),
                 },
                 resolve: function (env) {
                     validationLib.validateDslQuery(env);
-                    const queryResult = contentLib.query(createQueryDslParams(env, context));
+
+                    const start = env.args.after ? parseInt(graphQlConnectionLib.decodeCursor(env.args.after)) + 1 : 0;
```
Comment (rymsha) [inline src/main/resources/lib/guillotine/query/content-api.js:236]: Can after be anything but int?

File: src/main/resources/lib/guillotine/generic/generic-types.js [hunk 2]
```diff
@@ -524,6 +524,9 @@ function createGenericTypes(context) {
             aggregationsAsJson: {
                 type: graphQlLib.Json,
             },
+            highlightAsJson: {
+                type: graphQlLib.Json,
```
Comment (ComLock) [inline src/main/resources/lib/guillotine/generic/generic-types.js:528]: I guess thats the only way :)

I did the same

https://github.com/enonic/app-patchman/blob/952317ed2ce7c90d27c54545368e6b7c811d7951/src/main/resources/admin/tools/patchman/graphql/queries/addMultiRepoQuery.ts#L278

https://github.com/enonic/lib-explorer/blob/de5e6ca590213c71c20c3d03beedfb146a171425/src/main/resources/lib/explorer/interface/graphql/output/addOutputFieldsSearchResultHit.ts#L21

https://github.com/enonic/app-explorer/blob/3365ef9fed0c797c1e7c1c9579f1f222dd6cec96/src/main/resources/services/graphQL/document/addQueryDocuments.ts#L711

https://github.com/enonic/app-explorer/blob/52302cc45788110dbf6fb032231a94ed2fb4c8f6/src/main/resources/services/graphQL/synonym/addSynonymTypes.ts#L107

File: src/main/resources/lib/guillotine/generic/input-types.js [hunk 3]
```diff
@@ -627,6 +627,72 @@ function createInputTypes(context) {
...
+    const highlightCommonFields = createHighlightCommonFields(context);
+
+    context.types.highlightPropertiesInputType = createHighlightPropertiesInputType(context, highlightCommonFields);
+
+    context.types.highlightInputType = createHighlightInputType(context, highlightCommonFields);
+}
+
+function createHighlightCommonFields(context) {
+    return {
+        encoder: {
```
Comment (ComLock) [inline src/main/resources/lib/guillotine/generic/input-types.js:640]: I kept encoder and tagsSchema outside the common fields

https://developer.enonic.com/docs/xp/stable/storage/highlighting
Highlighting settings can be set on a global level and then overridden on the property level (except tagsSchema and encoder, these two can be set only on the global layer).

https://github.com/enonic/app-patchman/blob/952317ed2ce7c90d27c54545368e6b7c811d7951/src/main/resources/admin/tools/patchman/graphql/inputTypes/highlight/addInputTypeHighlight.ts#L36

File: src/test/resources/lib/factory-highlight-test.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        fragmentSize: 2,
+        noMatchSize: 2,
+        numberOfFragments: 5,
+        order: 'none',
+        preTag: '<section>',
+        postTag: '</section>',
+        requireFieldMatch: true,
+        properties: {
+            briefDescription: {
+                encoder: 'default',
```
Comment (ComLock) [inline src/test/resources/lib/factory-highlight-test.js:18]: This is illegal, encoder and tagSchema cannot be overridden...

**Review by ComLock — CHANGES_REQUESTED** [review]

File: src/main/resources/lib/guillotine/generic/input-types.js (hunk 3 above)
Comment (anatol-sialitski) [reply src/main/resources/lib/guillotine/generic/input-types.js]: Good catch!

**Review by rymsha — APPROVED** [review]

**Review by ComLock — APPROVED** [review]

**Review by vbradnitski — APPROVED** [review]

**Pull request merged**